runtime/common/src/main/kotlin/kotlinx/serialization/json/JsonElement.kt

```diff
@@ -146,25 +146,12 @@ public sealed class JsonPrimitive : JsonElement() {
  * and `JsonLiteral("42", true)` will be serialized as `42` and `"42"` respectively.
  */
 @Serializable(JsonLiteralSerializer::class) // TODO val for body is workaround for plugin bug
-public class JsonLiteral internal constructor(val body: Any, public val isString: Boolean) : JsonPrimitive() {
+sealed class JsonLiteral<T> private constructor(val body: T) : JsonPrimitive() {
```

Comment:
please do not remove explicit public modifier, it is crucial to have a proper understanding of what is and what is not public API (see [KEEP#45](https://github.com/Kotlin/KEEP/issues/45))